include/seqan3/alignment/pairwise/detail/policy_alignment_result_builder.hpp

```diff
@@ -0,0 +1,111 @@
+
+/*!\file
+ * \brief Provides seqan3::detail::policy_alignment_result_builder.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/alignment/pairwise/detail/type_traits.hpp>
+#include <seqan3/core/algorithm/configuration.hpp>
+#include <seqan3/core/detail/empty_type.hpp>
+#include <seqan3/core/type_traits/template_inspection.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Implements the alignment result builder.
+ * \ingroup pairwise_alignment
+ *
+ * \tparam alignment_configuration_t The type of the alignment configuration; must be a type specialisation of
+ *                                   seqan3::configuration.
+ *
+ * \details
+ *
+ * Implements the interfaces to build the alignment result based on the previously selected output configurations.
+ */
+template <typename alignment_configuration_t>
+#if !SEQAN3_WORKAROUND_GCC_93467
+//!\cond
+    requires is_type_specialisation_of_v<alignment_configuration_t, configuration>
+//!\endcond
+#endif // !SEQAN3_WORKAROUND_GCC_93467
+class policy_alignment_result_builder
+{
+protected:
+    //!\brief The configuration traits type.
+    using traits_type = alignment_configuration_traits<alignment_configuration_t>;
+    //!\brief The alignment result type.
+    using result_type = typename traits_type::alignment_result_type;
+
+    static_assert(!std::same_as<result_type, empty_type>, "The alignment result type was not configured.");
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    policy_alignment_result_builder() = default; //!< Defaulted.
+    policy_alignment_result_builder(policy_alignment_result_builder const &) = default; //!< Defaulted.
+    policy_alignment_result_builder(policy_alignment_result_builder &&) = default; //!< Defaulted.
+    policy_alignment_result_builder & operator=(policy_alignment_result_builder const &) = default; //!< Defaulted.
+    policy_alignment_result_builder & operator=(policy_alignment_result_builder &&) = default; //!< Defaulted.
+    ~policy_alignment_result_builder() = default; //!< Defaulted.
+
+    /*!\brief Construction and initialisation using the alignment configuration.
+     * \param[in] config The alignment configuration [not used in this context].
+     */
+    policy_alignment_result_builder(alignment_configuration_t const & SEQAN3_DOXYGEN_ONLY(config))
+    {}
+    //!\}
+
+    /*!\brief Builds the seqan3::alignment_result based on the given alignment result type and then invokes the
+     *        given callable with the result.
+     *
+     * \tparam indexed_sequence_pair_t The type of the indexed sequence pair.
+     * \tparam score_t The type of the score.
+     * \tparam callback_t The type of the callack to invoke.
+     *
+     * \param[in] indexed_sequence_pair The indexed sequence pair [maybe unused].
+     * \param[in] score The best alignment score [maybe unused].
```

Comment:
```suggestion
     * \param[in] indexed_sequence_pair The indexed sequence pair.
     * \param[in] score The best alignment score.
```

We don't use it anywhere. Why should a user care if we may not use that parameter, he has to provide that anyway to call it.

Reply:
🤠